srcs/python/kungfu/tensorflow/v1/optimizers/async_sgd.py

```diff
@@ -89,11 +94,17 @@ def apply_gradients(self, grads_and_vars, **kwargs):
                 with tf.control_dependencies([save_model_op]):
                     return tf.group(apply_op)
 
-    def distributed_initializer(self):
-        bcast_ops = []
-        for v in self.variables():
-            bcast_ops.append(tf.assign(v, broadcast(v)))
+    def _distributed_initializer(self):
+        bcast_ops = [tf.assign(v, broadcast(v)) for v in tf.global_variables()]
+
+        # FIXME: tf.trainable_variables() will return a TFOptimizer/iterations:0 if with Keras
```

Comment:
@lgarithm @marwage It seems like put a tf.train.optimizer into a Keras model.fit can cause this unexpected behavrious. In the Keras context, tf.trainable_variables may return variables that are actually not trainable. In the apply_gradients, the variables we intercept are correct.